### DISCUSSION TITLE
Warning when re-solving a model

### DISCUSSION THREAD

**Discussion by userdev163** [google/or-tools#4448] [CP-SAT questions]
Im sorry but it is very difficult to put code examples for my issue.  I'm using ortools 9.11. I have the following workflow in my project:
I create a scheduling model and add constraints (no overlap) until the problem is not solvable. 
Then I reuse the last feasible model that I saved (using a DeepCopy) and I solve this model again adding noOverlap constraints from another source. This raises : 
"Warning, an interval constraint was likely used without a corresponding linear constraint linking its start, size and end."
When I solve the model (cp_model) again via:
`solver = cp_model.CpSolver()`
`solver.Solve(self.model)`
I have tried to solve again adding no constraints and I have the same result. 
My best guess for know is that I have a constraint with a start value at 0, which would be a problem when reloaded (when exported to file this start becomes empty)
What could I be doing wrong in this situation?

**Comment by sschnug** [comment]
Did you follow [sat/docs/model.md](https://github.com/google/or-tools/blob/ed94162b910fa58896db99191378d3b71a5313af/ortools/sat/docs/model.md#model-copy) -> the variable-mapping step (`get_int_var_from_proto_index`)?

**Comment by lperron** [comment]
In 9.11 creating an interval also adds the linear constraint that enforces the start +size==end
 
This has been removed. 

Still I do not know why it fails.

Note that copying is tricky and error prone. I recommend rebuilding the model

**Comment by userdev163** [comment]
I have narrowed down my issue, it seems that the problem appears on my model because of a variable with a start = 0
When I'm giving an hint solution including this variable, the warning is generated. But when I skip this variable in the hint solution, no warning. 
I have also noticed that the solveur seems to remove this start = 0 and replace it with {} (when i print the model)
I think this may be related.

**Comment by lperron** [comment]
printing: this is the default behavior. 0 is the default value and is not
printed.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00



Le mar. 26 nov. 2024 à 09:47, userdev163 ***@***.***> a
écrit :
